internal/checkers/compares.go

```diff
@@ -42,13 +46,24 @@ func (checker Compares) Check(pass *analysis.Pass, call *CallMeta) *analysis.Dia
 		return nil
 	}
 
+	_, ptrComparison := pass.TypesInfo.TypeOf(be.X).(*types.Pointer)
+
 	var tokenToProposedFn map[token.Token]string
 
 	switch call.Fn.NameFTrimmed {
 	case "True":
-		tokenToProposedFn = tokenToProposedFnInsteadOfTrue
+		if ptrComparison {
+			tokenToProposedFn = tokenToProposedFnInsteadOfTrueForPtr
+		} else {
+			tokenToProposedFn = tokenToProposedFnInsteadOfTrue
+		}
+
 	case "False":
-		tokenToProposedFn = tokenToProposedFnInsteadOfFalse
+		if ptrComparison {
```

Comment:
This is the kind of implementation I was thinking for promoting zero over empty

https://github.com/Antonboom/testifylint/issues/75#issuecomment-2041380821

https://github.com/Antonboom/testifylint/issues/75#issuecomment-2067622523

Reply:
I have a "zero thread" in to-do :sorry:
complex story 😢

Reply:
OK, understood. I'm not very available either, haha